README.md
# A Privacy Policy Package for Laravel Applications<!-- omit in toc -->

[![Build Status](https://app.travis-ci.com/webflorist/privacy-policy-laravel.svg?branch=main)](https://app.travis-ci.com/github/webflorist/privacy-policy-laravel)
[![Latest Stable Version](https://poser.pugx.org/webflorist/privacy-policy-laravel/v)](https://packagist.org/packages/webflorist/privacy-policy-laravel)
[![License](https://poser.pugx.org/webflorist/privacy-policy-laravel/license)](https://github.com/webflorist/privacy-policy-laravel/blob/main/LICENSE.md)

This is a **Laravel** package providing an **open source** privacy policy available in **german** and **english**.

## Table of Contents<!-- omit in toc -->

- [Demo](#demo)
- [Features](#features)
- [Requirements](#requirements)
- [Installation](#installation)
- [Configuration](#configuration)
  - [`data-controller`: array (mandatory)](#data-controller-array-mandatory)
  - [`singular`: boolean (mandatory)](#singular-boolean-mandatory)
  - [`data-processing`: array (optional)](#data-processing-array-optional)
  - [`cookies`: array|false (mandatory)](#cookies-arrayfalse-mandatory)
  - [`processors`: array (optional)](#processors-array-optional)
- [Component Slots](#component-slots)
- [Disclaimer](#disclaimer)
- [License](#license)

## Demo

An demo application using the sister-package [webflorist/privacy-policy-vue](https://github.com/webflorist/privacy-policy-vue) with the same texts is avaliable at:  
<https://privacy-policy-vue-demo.netlify.app/>

## Features

- **Languages**  
  Currently the package includes texts in **german** and **english** language.

- **Singular/Plural**  
  You can choose between a singular or plural point of view.  
  (e.g. `My website...` vs `Our website...`)

- **Included Texts**

  - A general **intro text**
  - Listing of **GDPR rights**
  - Introduction of **data controller**
  - General **data security** text (SSL, etc.)
  - **Cookies** information
  - Information on **data processing** of third party data processors:
    - Webhosting
    - Web analytics
    - Interactive maps
    - Sending of emails (e.g. contact forms)
  - Disclaimer regarding **outgoing links**
  - Listing of all used **processors**

## Requirements

This package supports [`Laravel`](https://laravel.com/) from version 5.5 up until 8.x.

## Installation

1. Require the package via composer:  
   `composer require webflorist/privacy-policy-laravel`
2. Publish config:  
   `php artisan vendor:publish --provider="Webflorist\PrivacyPolicy\PrivacyPolicyServiceProvider"`
3. Configure your privacy policy with the `privacy-policy.php` file in Laravel's `config` folder. (see below for details).
4. Include the privacy policy component in your view:

```blade
    @component('webflorist-privacy-policy::privacy-policy')
    @endcomponent
```

## Configuration

### `data-controller`: array (mandatory)

Contact info for the data controller. The array can have the following elements:

```php
[
 'organisation' => 'Acme Corporation',
 'name' => 'John Doe',
 'address' => 'Acme Street 1, 123456 Acme City, USA',
 'email' => 'privacy@example.com',
 'phone' => '+1 555-0123',
];
```

### `singular`: boolean (mandatory)

Set to `true` for singular viewpoint (e.g. "My website...").

Set to `false` for plural viewpoint (e.g. "Our website...").

### `data-processing`: array (optional)

The data processings used by your site. Each process must be stated as two-dimenstional array with the type of process as it's key and it's properties stated as a value-array.

Supported types of processes are:

- `webhosting`
- `analytics`
- `maps`
- `send_emails`

The properties of a process can be:

- `processor`: String (mandatory)  
  The key of the processor providing this processing (either one of the [included ones](https://github.com/webflorist/privacy-policy-text/blob/main/src/processors.php) or one stated via the [processors property](#processors-array-optional))
- `service`: String (optional)  
  Name of the service providing this processing (e.g. `Google Analytics` or `Google Maps`)
- `data_categories`: Array (mandatory)  
  Array of data categories processed by this process. Supported values are:
  - `inventory_data`
  - `usage_data`
  - `geo_data`
  - `usage_statistics`
  - `contract_data`
  - `payment_data`

Here is an example for a full `data_processing` config:

```php
[
 'webhosting' => [
  'processor' => ['netlify', 'storyblok'],
  'data_categories' => ['usage_data'],
 ],
 'analytics' => [
  'processor' => 'google_eu',
  'service' => 'Google Analytics',
  'data_categories' => ['usage_data', 'usage_statistics'],
 ],
 'maps' => [
  'processor' => 'google_usa',
  'service' => 'Google Maps',
  'data_categories' => ['usage_data', 'geo_data'],
 ],
 'send_emails' => [
  'processor' => 'twilio_eu',
  'service' => 'Twilio Sendgrid',
  'data_categories' => ['usage_data', 'inventory_data'],
 ],
];
```

### `cookies`: array|false (mandatory)

This config describes the cookies used by your site.

If your site uses no cookies at all, simply set this to `false`.

If not, divide the used cookies into first party cookies and third party ones, and list them as sub-arrays of the `first_party` and `third_party` arrays of the `cookies` config.

Each cookie is described as an array with the following possible properties:

- `name`: String (mandatory)  
  The name of the cookie
- `purpose`: String (mandatory)  
  The key of the cookie purpose. Can be one of the following:
  - `session`: Session cookie
  - `xsrf`: Cookie to prevent "Cross-Site Request Forgery" attacks
  - `hide_alert`: Cookie to prevent displaying the cookie dialog again after hiding it
  - `all_choices`: Cookie storing the choices regarding various cookies displayed in the cookie dialog
  - `analytics_choice`: Cookie storing the choice regarding the usage of web analytics in the cookie dialog
  - `maps_choice`: Cookie storing the choice regarding the usage of interactive maps
  - `analytics_third_party`: Cookies written by the web analytics tool
  - `maps_third_party`: Cookies set on displaying interactive maps.
- `written_on`: String (mandatory)  
  When the cookie is created. Can be one of the following:
  - `every_visit`: Written on every visit
  - `hide_alert`: Written on hiding the cookie dialog
  - `maps`: Written on acknowledging the usage of interactive maps
  - `accept_cookies`: Written on accepting the corresponding cookies
- `duration`: String (mandatory)  
  The duration of the cookie. Can be one of the following:
  - `end_of_session`
  - `1_year`
  - `2_years`
  - `24_hours`
  - `1_minute`
  - `various`
- `processor`: String (mandatory with third party cookies)
  The key of the processor providing this processing (either one of the [included ones](https://github.com/webflorist/privacy-policy-text/blob/main/src/processors.php) or one stated via the [processors property](#processors-array-optional))

Here is an example of the cookie config:

```php
[
 'first_party' => [
  [
   'name' => 'session',
   'purpose' => 'session',
   'written_on' => 'every_visit',
   'duration' => 'end_of_session',
  ],
 ],
 'third_party' => [
  [
   'name' => '_ga, _gat, _gid',
   'purpose' => 'analytics_third_party',
   'written_on' => 'accept_cookies',
   'duration' => 'various',
   'processor' => 'google_eu',
   'service' => 'Google Analytics',
  ],
 ],
];
```

### `processors`: array (optional)

Definition of processors stated with [data processings](#data-processing-array-optional) or [cookies](#cookies-arrayfalse-mandatory).

Several processors are [already included](https://github.com/webflorist/privacy-policy-text/blob/main/src/processors.php).

State your own ones in the processors config in an array with the shorthand-key of the processor as the key and it's info as an array with the following elements:

- `name` String (mandatory)  
  Company name
- `address` String (mandatory)  
  Full address of the company
- `privacy_policy` String (mandatory)
  Link to the processors privacy policy
- `privacy_shield` String (optional)  
  Link to the [privacy shield](https://www.privacyshield.gov) entry of the processor.

Here is and example:

```php
[
 'acme_corp' => [
  'name' => 'Acme Corporation',
  'address' => 'Acme Street 1, 123456 Acme City, USA',
  'privacy_policy' => 'https://www.example.com/privacy',
  'privacy_shield' =>
   'https://www.privacyshield.gov/participant?id=a2zt0000000TOWQAA4',
 ],
];
```

## Component Slots

The `webflorist-privacy-policy::privacy-policy` component provides the following named slots to insert custom text on various positions:

| Slot Name                         | Usage                                                             |
| --------------------------------- | ----------------------------------------------------------------- |
| after_intro                       | `@slot('after_intro') Custom Text @endslot`                       |
| gdpr_rights_start                 | `@slot('gdpr_rights_start') Custom Text @endslot`                 |
| gdpr_rights_end                   | `@slot('gdpr_rights_end') Custom Text @endslot`                   |
| data_controller_start             | `@slot('data_controller_start') Custom Text @endslot`             |
| data_controller_end               | `@slot('data_controller_end') Custom Text @endslot`               |
| security_start                    | `@slot('security_start') Custom Text @endslot`                    |
| security_end                      | `@slot('security_end') Custom Text @endslot`                      |
| cookies_start                     | `@slot('cookies_start') Custom Text @endslot`                     |
| cookies_end                       | `@slot('cookies_end') Custom Text @endslot`                       |
| data_processing_start             | `@slot('data_processing_start') Custom Text @endslot`             |
| data_processing_webhosting_start  | `@slot('data_processing_webhosting_start') Custom Text @endslot`  |
| data_processing_webhosting_end    | `@slot('data_processing_webhosting_end') Custom Text @endslot`    |
| data_processing_analytics_start   | `@slot('data_processing_analytics_start') Custom Text @endslot`   |
| data_processing_analytics_end     | `@slot('data_processing_analytics_end') Custom Text @endslot`     |
| data_processing_maps_start        | `@slot('data_processing_maps_start') Custom Text @endslot`        |
| data_processing_maps_end          | `@slot('data_processing_maps_end') Custom Text @endslot`          |
| data_processing_send_emails_start | `@slot('data_processing_send_emails_start') Custom Text @endslot` |
| data_processing_send_emails_end   | `@slot('data_processing_send_emails_end') Custom Text @endslot`   |
| data_processing_end               | `@slot('data_processing_end') Custom Text @endslot`               |
| outgoing_links_start              | `@slot('outgoing_links_start') Custom Text @endslot`              |
| outgoing_links_end                | `@slot('outgoing_links_end') Custom Text @endslot`                |
| processor_list_start              | `@slot('processor_list_start') Custom Text @endslot`              |
| processor_list_end                | `@slot('processor_list_end') Custom Text @endslot`                |

## Disclaimer

The included text _should_ be suitable for a GDPR-compliant website.

**I however do not take any responsibility whatsowever for that.**

## License

This package is open-sourced software licensed under the [MIT license](https://github.com/webflorist/privacy-policy-laravel/blob/main/LICENSE.md).


CHANGELOG.md
# Release Notes

## [1.0.2](https://github.com/webflorist/privacy-policy-laravel/compare/v1.0.1...v1.0.2) (2022-01-15)


### Bug Fixes

* readme ([eb1c618](https://github.com/webflorist/privacy-policy-laravel/commit/eb1c61813af16be74ee878e69eed1e31a01b8f94))

## [v1.0.1 (2022-01-06)](https://github.com/webflorist/privacy-policy-text/compare/v1.0.0...v1.0.1)

- FIX html code in cookie details (a97d59a)


tests/TestCase.php
<?php

namespace PrivacyPolicyTests;

use Illuminate\Contracts\Config\Repository;
use Illuminate\Routing\Router;
use Orchestra\Testbench\TestCase as BaseTestCase;
use Webflorist\PrivacyPolicy\PrivacyPolicyServiceProvider;

class TestCase extends BaseTestCase
{
	protected function getPackageProviders($app)
	{
		return [PrivacyPolicyServiceProvider::class];
	}

	protected function getPackageAliases($app)
	{
		return [
			'PrivacyPolicy' => \Webflorist\PrivacyPolicy\Facades\PrivacyPolicy::class,
		];
	}

	protected function getEnvironmentSetUp($app)
	{
		$app['router']->view('/privacy-policy', 'privacy-policy');

		$app['config']->set('view.paths', [dirname(__FILE__) . '/Feature/Views']);

		$app['config']->set('app.locale', 'en');

		$app['config']->set('privacy-policy.data_processing', [
			'webhosting' => [
				'processor' => ['netlify', 'storyblok'],
				'data_categories' => ['usage_data'],
			],
			'analytics' => [
				'processor' => 'google_eu',
				'service' => 'Google Analytics',
				'data_categories' => ['usage_data', 'usage_statistics'],
			],
			'maps' => [
				'processor' => 'google_usa',
				'service' => 'Google Maps',
				'data_categories' => ['usage_data', 'geo_data'],
			],
			'send_emails' => [
				'processor' => 'twilio_eu',
				'service' => 'Twilio Sendgrid',
				'data_categories' => ['usage_data', 'inventory_data'],
			],
		]);

		$app['config']->set('privacy-policy.cookies', [
			'first_party' => [
				[
					'name' => 'session',
					'purpose' => 'session',
					'written_on' => 'every_visit',
					'duration' => 'end_of_session',
				],
			],
			'third_party' => [
				[
					'name' => '_ga, _gat, _gid',
					'purpose' => 'analytics_third_party',
					'written_on' => 'accept_cookies',
					'duration' => 'various',
					'processor' => 'google_eu',
					'service' => 'Google Analytics',
				],
			],
		]);
	}
}


tests/Feature/PrivacyPolicyTest.php
<?php

namespace PrivacyPolicyTests\Feature;

use PrivacyPolicyTests\TestCase;

class PrivacyPolicyTest extends TestCase
{
	public function test_privacy_policy()
	{
		$response = $this->get('privacy-policy');
		$response->assertSee(
			'The privacy and protection of your data is of great concern to us.'
		);
		$response->assertSee(
			'We handle your data in accordance with the European General Data Protection Regulation'
		);
		$response->assertSee('Acme Corporation');
		$response->assertSee(
			'Our website takes many technical measures to provide a secure transmission'
		);
		$response->assertSee('First Party Cookies');
		$response->assertSee('session');
		$response->assertSee('Third Party Cookies');
		$response->assertSee('_ga, _gat, _gid');
		$response->assertSee(
			'When visiting any website, your browser is automatically transmitting information to the server and infrastructure it is hosted on'
		);
		$response->assertSee(
			'If you accept web analytics in the cookie notification, our website will forward log data to the service'
		);
		$response->assertSee(
			'Our website provides the functionality of interactive maps on demand'
		);
		$response->assertSee(
			'Our website includes forms, that can be used to send us an e-mail.'
		);
		$response->assertSee(
			'Our website includes links to other websites, which are not operated by us'
		);
		$response->assertSee(
			'2325 3rd Street, Suite 296, San Francisco, CA 94107, USA'
		);
	}
}


tests/Feature/Views/privacy-policy.blade.php
<html>

<body>

    @component('webflorist-privacy-policy::privacy-policy')

    @slot('after_intro')
        Forbidden
    @endslot

    @endcomponent

</body>

</html>

release.sh
#!/usr/bin/env bash
bump=$1

valid_bumps=(major minor patch)

if [[ ! " ${valid_bumps[*]} " =~ " ${1} " ]]; then
	echo 'No version bump specified. Call either:'
	echo '$ ./release.sh major'
	echo '$ ./release.sh minor'
	echo '$ ./release.sh patch'
	exit 1
fi

# Check, if we are on 'main' branch
current_branch=$(git rev-parse --abbrev-ref HEAD)
if [[ $current_branch != 'main' ]]; then
	echo "Please switch to main branch before releasing a new version!"
	exit 1
fi

echo
echo "=============="
echo "Pulling branch"
echo "=============="
git pull
if [[ $? > 0 ]]; then exit 1; fi

# Check, if there are uncommited changes
git_status=$(git status)
if [[ $? > 0 ]]; then exit 1; fi
if [[ $git_status != *"nothing to commit, working tree clean"* ]]; then
	echo
	echo "You seem to have uncommited changes. Please commit them before releasing a new version!"
	exit 1
fi

echo
echo "============"
echo "Pulling tags"
echo "============"
git pull --tags
if [[ $? > 0 ]]; then exit 1; fi

echo
echo "===================="
echo "Getting last version"
echo "===================="
last_tag=$(git describe --tags --abbrev=0)
if [[ $? > 0 ]]; then exit 1; fi
old_version=${last_tag:1}
echo $old_version

echo
echo "======================="
echo "Calculating new version"
echo "======================="
IFS='.' read -r -a version_array <<< "$old_version"
if [[ $bump == 'major' ]]; then
	version_array[0]=$((version_array[0] + 1))
	version_array[1]=0
	version_array[2]=0
elif [[ $bump == 'minor' ]]; then
	version_array[0]=${version_array[0]}
	version_array[1]=$((version_array[1] + 1))
	version_array[2]=0
elif [[ $bump == 'patch' ]]; then
	version_array[0]=${version_array[0]}
	version_array[1]=${version_array[1]}
	version_array[2]=$((version_array[2] + 1))
fi
new_version=${version_array[0]}.${version_array[1]}.${version_array[2]}
echo $new_version

echo
echo "========="
echo "Changelog"
echo "========="
changelog=$(git log --pretty="- %s (%h)" v${old_version}...)
if [[ $? > 0 ]]; then exit 1; fi
echo "$changelog"

echo
read -p "Do you want to release this version? (y/n)" -n 1 -r
echo
if [[ $REPLY =~ ^[Yy]$ ]]; then

	echo
	echo "=========================="
	echo "Prepending to CHANGELOG.md"
	echo "=========================="
	changelog_header="# Release Notes"
	printf -v date '%(%Y-%m-%d)T' -1
	sed -i "s/${changelog_header}//" CHANGELOG.md
	new_changelog=$changelog_header
	new_changelog+=$'\n\n'
	new_changelog+="## [v${new_version} (${date})](https://github.com/webflorist/privacy-policy-text/compare/v${old_version}...v${new_version})"
	new_changelog+=$'\n\n'
	new_changelog+=$changelog
	new_changelog+=$(cat CHANGELOG.md)
	echo "$new_changelog" > CHANGELOG.md

	echo
	echo "=================="
	echo "Committing changes"
	echo "=================="
	git commit -a -m "RELEASE ${new_version}"
	if [[ $? > 0 ]]; then exit 1; fi

	echo
	echo "=============="
	echo "Pushing branch"
	echo "=============="
	git push
	if [[ $? > 0 ]]; then exit 1; fi

	echo
	echo "============"
	echo "Creating and pushing tag"
	echo "============"
	git tag -a v${new_version} -m "v${new_version}"
	if [[ $? > 0 ]]; then exit 1; fi
	git push --tags
	if [[ $? > 0 ]]; then exit 1; fi

fi


src/PrivacyPolicy/PrivacyPolicy.php
<?php

namespace Webflorist\PrivacyPolicy;

/**
 * Class PrivacyPolicy
 * @package Webflorist\PrivacyPolicy
 */
class PrivacyPolicy
{
	private array $allProcessors = [];

	private array $interpolations = [];

	private array $usedProcessors = [];

	/**
	 * PrivacyPolicy constructor.
	 */
	public function __construct()
	{
		$this->allProcessors = array_merge(
			config('privacy-policy.processors'),
			\Webflorist\PrivacyPolicyText\Helpers::getProcessors()
		);

		// Determine used processors and interpolations for translation
		foreach (
			config('privacy-policy.data_processing')
			as $processType => $process
		) {
			$processors = is_array($process['processor'])
				? $process['processor']
				: [$process['processor']];

			// Create interpolations for translation of texts.
			$processorLinks = [];
			foreach ($processors as $processorKey) {
				if (!isset($this->allProcessors[$processorKey])) {
					throw new \Webflorist\PrivacyPolicy\Exceptions\ProcessorNotFoundException(
						"Processor '$processorKey' used for data-processing '$processType' not found in processor-list. Please state processor details via the config key 'privacy-policy.processors'."
					);
				}
				$processorName = $this->allProcessors[$processorKey]['name'];
				$processorLinks[] =
					'<a href="#processor-' .
					$processorKey .
					'">' .
					$processorName .
					'</a>';
			}
			$this->interpolations[$processType . '_processor'] = implode(
				', ',
				$processorLinks
			);
			$this->interpolations[$processType . '_service'] =
				$process['service'] ??
				__(
					"Webflorist-PrivacyPolicy::privacy-policy.data_processing.$processType.title"
				);

			// Put processors in $this->usedProcessors
			foreach ($processors as $processorKey) {
				if (!isset($this->usedProcessors[$processorKey])) {
					$this->usedProcessors[$processorKey] =
						$this->allProcessors[$processorKey];
				}

				// Add data purpose to processor.
				if (!isset($this->usedProcessors[$processorKey]['purposes'])) {
					$this->usedProcessors[$processorKey]['purposes'] = [];
				}
				if (
					!in_array(
						$processType,
						$this->usedProcessors[$processorKey]['purposes']
					)
				) {
					$this->usedProcessors[$processorKey]['purposes'][] = $processType;
				}

				// Add data categories to processor.
				if (!isset($this->usedProcessors[$processorKey]['data_categories'])) {
					$this->usedProcessors[$processorKey]['data_categories'] = [];
				}
				$this->usedProcessors[$processorKey]['data_categories'] = array_unique(
					array_merge(
						$this->usedProcessors[$processorKey]['data_categories'],
						$process['data_categories']
					)
				);
			}
		}
	}

	public function getUsedProcessors()
	{
		return $this->usedProcessors;
	}

	public function translate(string $key)
	{
		return \Webflorist\PrivacyPolicyText\Helpers::renderText(
			__("Webflorist-PrivacyPolicy::privacy-policy.$key", $this->interpolations)
		);
	}
}


src/resources/views/cookie-details.blade.php
<section>
    <h4>{{ $cookie['name'] }}</h4>

    <dl>
        <dt>{!! \PrivacyPolicy::translate('cookies.purpose.title') !!}</dt>
        <dd>{!! \PrivacyPolicy::translate('cookies.purpose.' . $cookie['purpose']) !!}</dd>

        @if ($type === 'third_party')
            <dt>{!! \PrivacyPolicy::translate('cookies.origin') !!}</dt>
            <dd>
                {{ $cookie['service'] }}
                (<a href="#processor-{{ $cookie['processor'] }}">
                    {{ $processors[$cookie['processor']]['name'] }}
                </a>)
            </dd>
        @endif

        <dt>{!! \PrivacyPolicy::translate('cookies.written_on.title') !!}</dt>
        <dd>{!! \PrivacyPolicy::translate('cookies.written_on.' . $cookie['written_on']) !!}</dd>

        <dt>{!! \PrivacyPolicy::translate('cookies.duration.title') !!}</dt>
        <dd>{!! \PrivacyPolicy::translate('cookies.duration.' . $cookie['duration']) !!}</dd>
    </dl>
</section>


src/PrivacyPolicy/PrivacyPolicyServiceProvider.php
<?php

namespace Webflorist\PrivacyPolicy;

use Illuminate\Foundation\Http\Kernel;
use Illuminate\Routing\Router;
use Illuminate\Support\ServiceProvider;

class PrivacyPolicyServiceProvider extends ServiceProvider
{
	/**
	 * Register the application services.
	 *
	 * @return void
	 */
	public function register()
	{
		$this->mergeConfig();
		$this->registerService();
	}

	/**
	 * Bootstrap the application services.
	 *
	 * @return void
	 */
	public function boot()
	{
		$this->publishConfig();
		$this->loadTranslations();
		$this->loadViews();
	}

	protected function mergeConfig()
	{
		$this->mergeConfigFrom(
			__DIR__ . '/../config/privacy-policy.php',
			'privacy-policy'
		);
	}

	protected function registerService()
	{
		$this->app->singleton(PrivacyPolicy::class, function () {
			return new PrivacyPolicy();
		});
	}

	protected function publishConfig()
	{
		$this->publishes([
			__DIR__ . '/../config/privacy-policy.php' => config_path(
				'privacy-policy.php'
			),
		]);
	}

	private function loadTranslations()
	{
		$vendorDir = $this->app->basePath('vendor');
		if ($this->app->environment('testing')) {
			$vendorDir = __DIR__ . '/../../vendor';
		}
		$viewpoint = config('privacy-policy.singular') ? 'singular' : 'plural';
		$this->loadTranslationsFrom(
			$vendorDir .
				"/webflorist/privacy-policy-text/dist/php/colon-prefix/$viewpoint",
			'Webflorist-PrivacyPolicy'
		);
	}

	private function loadViews()
	{
		$this->loadViewsFrom(
			__DIR__ . '/../resources/views',
			'webflorist-privacy-policy'
		);
	}

	private function addMiddleware()
	{
		/** @var Kernel $kernel */
		$kernel = $this->app['Illuminate\Contracts\Http\Kernel'];
		$kernel->pushMiddleware(RouteTreeMiddleware::class);

		/** @var Router $router */
		$router = $this->app['router'];
		if ($router->hasMiddlewareGroup('web')) {
			$router->pushMiddlewareToGroup('web', SessionLocaleMiddleware::class);
		}
	}

	private function addRoutes()
	{
		/** @var Router $router */
		$router = $this->app['router'];

		if (config('routetree.api.enabled')) {
			$router->group(
				['prefix' => config('routetree.api.base_path'), 'middleware' => 'api'],
				function (Router $router) {
					$router
						->resource('routes', RoutesController::class)
						->only(['index', 'show']);
				}
			);
		}

		if (config('routetree.sitemap.route.enabled')) {
			$router->get(config('routetree.sitemap.route.path'), [
				XmlSitemapController::class,
				'get',
			]);
		}
	}
}


src/config/privacy-policy.php
<?php

return [
	/*
    |--------------------------------------------------------------------------
    | Singular or Plural Viewpoint
    |--------------------------------------------------------------------------
    |
    | Set to true for singular viewpoint (e.g. "My website...").
    | Set to false for plural viewpoint (e.g. "Our website...").
    |
    */
	'singular' => false,

	/*
    |--------------------------------------------------------------------------
    | Data Controller
    |--------------------------------------------------------------------------
    |
    | Contact data of the data controller.
    |
    */
	'data_controller' => [
		'organisation' => 'Acme Corporation',
		'name' => 'John Doe',
		'address' => 'Acme Street 1, 123456 Acme City, USA',
		'email' => 'privacy@example.com',
		'phone' => '+1 555-0123',
	],

	/*
    |--------------------------------------------------------------------------
    | Data Processing
    |--------------------------------------------------------------------------
    |
    | For details see https://github.com/webflorist/privacy-policy-laravel#data-processing-array-optional
    |
    */
	'data_processing' => [],

	/*
    |--------------------------------------------------------------------------
    | Cookies
    |--------------------------------------------------------------------------
    |
    | This config describes the cookies used by your site.
    |
    | If your site uses no cookies at all, simply set this to `false`.
    |
    | If not, see https://github.com/webflorist/privacy-policy-laravel#cookies-arrayfalse-mandatory
    |
    */
	'cookies' => false,

	/*
    |--------------------------------------------------------------------------
    | Processors
    |--------------------------------------------------------------------------
    |
    | Definition of processors used with data processings or cookies.
    |
    | Several processors are already included.
    | See: https://github.com/webflorist/privacy-policy-text/blob/main/src/processors.php
    |
    | On the configuration of additional ones see:
    | https://github.com/webflorist/privacy-policy-laravel#processors-array-optional
    |
    */
	'processors' => [],
];
